plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -358,107 +358,114 @@ bool CppParser::preparse()
                    }
                  });
 
-  try
+  // Detect changed files through C++ header inclusions.
+  util::OdbTransaction {_ctx.db} ([&]
   {
-    util::OdbTransaction {_ctx.db} ([&]
+    for (const model::FilePtr file : filePtrs)
     {
-      for (const model::FilePtr& file : filePtrs)
+      if(file)
       {
-        if(file)
-        {
-          markAsModified(file);
-        }
+        markByInclusion(file);
       }
+    }
+  }); // end of transaction
 
-      for (auto& item : _ctx.fileStatus)
+  // Perform maintenance actions.
+  if(!dry_)
```

Comment:
We could do this:

```.cpp
if (dry_)
  return true;
```

This way the rest of the file doesn't have to be modified and indented. Sorry, I'm not a big fan of too deeply nested and indented stuff.

Reply:
Done.